api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
In theory, `events` can report container's removal to us, so it's possible. But then we'll need two API calls: `WaitStop` and `WaitRemoveFromEvents`, it won't be more elegant than doing both in single `Wait` API.

And actually I am getting the exit code through watching events, and some of us don't quite like it, so I guess it won't satisfy all of us even we waitForRemoval with `Events` API. :crying_cat_face: 
 
I think it's OK to wait container to remove with an extra arguments of `ContainerWait` personally, it still makes sense.